tutorials/java/chapters/ch06.adoc

```diff
@@ -0,0 +1,191 @@
+
+And then it will cover how to test such applications.
+
+### Deploy Using Pre-Built WAR
+
+Pull the Docker image that contains WildFly and pre-built Java EE 7 application WAR file as shown:
+
+[source, text]
+----
+docker pull arungupta/javaee7-hol
+----
+
+The https://github.com/arun-gupta/docker-images/blob/master/javaee7-hol/Dockerfile[javaee7-hol Dockerfile] is based on `jboss/wildfly` and adds the movieplex7 application as war file.
+
+Run it:
+
+[source, text]
+----
+docker run -it -p 8080:8080 arungupta/javaee7-hol
+----
+
+See the application in action at http://dockerhost:8080/movieplex7/. The output is shown:
+
+.Java EE 7 Application Output
+image::javaee7-movieplex7.png[]
+
+This uses an in-memory database with WildFly application server as shown in the image:
+
+.In-memory Database
+image::javaee7-hol-in-memory-database.png[]
+
+Only two changes are required to the standard `jboss/wildfly` image:
+
+. By default, WildFly starts in Web platform. This Java EE 7 application uses some capabilities from the Full Platform and so WildFly is started in that mode instead as:
++
+[source, text]
+----
+CMD ["/opt/jboss/wildfly/bin/standalone.sh", "-c", "standalone-full.xml", "-b", "0.0.0.0"]
+----
++
+. WAR file is copied to the `standalone/deployments` directory as:
++
+[source, text]
+----
+RUN curl -L https://github.com/javaee-samples/javaee7-hol/raw/master/solution/movieplex7-1.0-SNAPSHOT.war -o /opt/jboss/wildfly/standalone/deployments/movieplex7-1.0-SNAPSHOT.war
+----
+
+### Deploy Using Container Linking
+
+<<JavaEE7_PreBuilt_WAR>> explained how to use an in-memory database with the application server. This gets you started rather quickly but becomes a bottleneck soon as the database is only in-memory. This means that any changes made to your schema and data are lost when the application server shuts down. In this case, you need to use a database server that resides outside the application server. For example, MySQL as the database server and WildFly as the application server.
+
+.Two Containers On Same Docker Host
+image::javaee7-hol-container-linking.png[]
+
+This section will show how https://docs.docker.com/userguide/dockerlinks/[Docker Container Linking] can be used to connect to a service running inside a Docker container via a network port.
+
+. Start MySQL server as:
++
+[source, text]
+----
+docker run --name mysqldb -e MYSQL_USER=mysql -e MYSQL_PASSWORD=mysql -e MYSQL_DATABASE=sample -e MYSQL_ROOT_PASSWORD=supersecret -p 3306:3306 -d mysql
+----
++
+`-e` define environment variables that are read by the database at startup and allow us to access the database with this user and password.
++
+. Start WildFly and deploy Java EE 7 application as:
++
+[source, text]
+----
+docker run -it --name mywildfly --link mysqldb:db -p 8080:8080 arungupta/wildfly-mysql-javaee7
+----
++
+`--link` takes two parameters - first is name of the container we're linking to and second is the alias for the link name.
```

Comment:
SGTM -- I think the faster the user can progress through the content to getting their app all up and running, the more rewarding the Docker experience will be. :)  So I'm +1 on de-duplicating content!